sources/log_exp_math.move

```diff
@@ -176,18 +178,70 @@ module fixed_point64::log_exp_math {
     }
 
     public fun pow(x: FixedPoint64, y: FixedPoint64): FixedPoint64 {
-        let result;
-        if (fixed_point64::to_u128(y) == 0) {
-            // We solve the 0^0 indetermination by making it equal to one.
-            result = fixed_point64::one();
-        } else if (fixed_point64::to_u128(x) == 0) {
-            result = fixed_point64::zero();
+        let (success, result) = try_simple_pow(x, y);
+        if (success) {
+            result
         } else {
             // x^y = exp(y * ln(x))
             let (sign, ln_x) = ln(x);
             let y_times_ln_x = fixed_point64::mul_fp(y, ln_x);
-            result = exp(sign, y_times_ln_x);
-        };
-        result
+            exp(sign, y_times_ln_x)
+        }
+    }
+    
+    /// pow_up adds 10^-9 to the result of pow if Taylor series is used
+    /// so that the result is always greater than or equal to the true value
+    public fun pow_up(x: FixedPoint64, y: FixedPoint64): FixedPoint64 {
+        let (success, result) = try_simple_pow(x, y);
+        if (success) {
+            result
+        } else {
+            // x^y = exp(y * ln(x))
+            let (sign, ln_x) = ln(x);
+            let y_times_ln_x = fixed_point64::mul_fp(y, ln_x);
+            fixed_point64::add_fp(exp(sign, y_times_ln_x), fixed_point64::from_u128(TEN_EXP_MINUS_9))
+        }
+    }
+    
+    /// pow_down substracts 10^-9 from the result of pow if Taylor series is used and the result is greater than 10^-9
+    /// so that the result is always smaller than or equal to the true value
```

Comment:
> so that the result is always smaller than or equal to the true value

Is this proven? I remember there was underestimation when doing (MAX_U64/4)^4 != MAX_U64. I wonder with `pow_up` if it works now?

Reply:
It's not proven. I tested with (MAX_U64 ^ (1/8))^8. `pow_down` is not working if using substraction of 1e-9. we have to use multiplication by (1 +/- 1e-9) to make it working, so I changed the implementation